addons/udes_stock_refactoring/tests/test_splitting.py

```diff
@@ -0,0 +1,659 @@
+
+    def test_items_added_to_existing_picking(self):
+        """Create two pickings for two items on the same pallet. Reserve them
+        sequentially and check they result in one picking with two moves.
+        """
+        pallet = self.create_package()
+
+        self.create_quant(
+            self.elderberry.id, self.test_stock_location_01.id, 5, package_id=pallet.id
+        )
+        self.create_quant(
+            self.elderberry.id, self.test_stock_location_01.id, 10, package_id=pallet.id
+        )
+
+        pick1 = self.picking
+        pick2 = self.create_picking(self.picking_type_pick)
+
+        move1 = self.create_move(self.elderberry, 5, pick1)
+        move2 = self.create_move(self.elderberry, 10, pick2)
+
+        pick1.action_assign()
+        pick2.action_assign()
+
+        pick = (move1 | move2).picking_id
+
+        self.assertEqual(len(pick), 1)
+        self.assertEqual(pick.state, "assigned")
+        self.assertEqual(pick.group_id.name, pallet.name)
+
+        # Check we haven't mangled the moves or move_lines
+        self.assertEqual((move1 | move2).ids, pick.move_lines.ids)
+
+        move_lines = pick.move_line_ids
+        self.assertEqual(len(move_lines), 2)
+
+        move_line1 = move_lines.filtered(lambda ml: ml.move_id == move1)
+        self.assertEqual(move_line1.product_qty, 5)
+        self.assertEqual(move_line1.package_id, pallet)
+        self.assertEqual(move_line1.product_id, self.elderberry)
+
+        move_line2 = move_lines.filtered(lambda ml: ml.move_id == move2)
+        self.assertEqual(move_line2.product_qty, 10)
+        self.assertEqual(move_line2.package_id, pallet)
+        self.assertEqual(move_line2.product_id, self.elderberry)
+
+        # Check that the first picking has been reused and the second picking
+        # has been marked as empty (u_mark = False) and deleted at the very end
+        # of the action
+        self.assertTrue(pick1.exists())
+        self.assertFalse(pick2.exists())
+
+    def test_check_non_default_locations_maintained(self):
+        """Reserve when the locations of the picking are not the defaults of
+        the picking type and check the non-defaults are maintained."""
+        self.picking.write(
+            {
+                "location_id": self.test_stock_location_01.id,
+                "location_dest_id": self.test_goodsout_location_01.id,
+            }
+        )
+
+        apple_pallet = self.create_package()
+        self.create_quant(
+            self.apple.id, self.test_stock_location_01.id, 10, package_id=apple_pallet.id
+        )
+        apple_move = self.create_move(self.apple, 10, self.picking)
+
+        self.picking.action_assign()
+
+        apple_pick = apple_move.picking_id
+        self.assertEqual(self.picking, apple_pick)  # Check picking reuse
+        self.assertEqual(apple_pick.state, "assigned")
+        self.assertEqual(apple_pick.location_id.id, self.test_stock_location_01.id)
```

Comment:
Don't need the id's

Reply:
Good spot, have changed this and a few other occasions this happens in these unit tests.